### PR TITLE
superlu_dist: update to 4.2

### PR DESCRIPTION
**Pull request by davydden**

### CODE REVIEW

**Comment by dpo** [comment]
https://github.com/Homebrew/homebrew/pull/46461

File: superlu_dist.rb [hunk 1]
```diff
@@ -21,7 +21,7 @@ class SuperluDist < Formula
   # fix duplicate symbols [mc64dd_,mc64ed_,mc64fd_] when linking with superlu
   patch do
     url "https://bitbucket.org/petsc/pkg-superlu_dist/commits/2faf8669a2ba20250ffe2d8a1b63d4f8ef8c5b74/raw/"
-    sha256 "09dcd7de83ef9d2590465742df76753cb7c61bfe8e7819c02613d9bcf20ed255"
+    sha256 "67e2966e1a9b4e3471374d1e720946c2ef34403d00e796767b51718e4d50604f"
```
Comment (dpo) [inline superlu_dist.rb]: See #3098

Comment (davydden) [inline superlu_dist.rb]: i know, but it's not yet merged. When you merge, i can rebase

Comment (dpo) [inline superlu_dist.rb]: Ok, you can rebase.

**Pull request closed without merging**